ntbs-service/Migrations/20191022104529_UpdatePostcodeLookupTable.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.IO;
+using CsvHelper;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace ntbs_service.Migrations
+{
+    public partial class UpdatePostcodeLookupTable : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            var pathToFile = Path.Combine(Environment.CurrentDirectory, "Models/SeedData/Postcodes.csv");
+            const int itemsPerUpdate = 1000;
+
+            var postcodeArray = new string[itemsPerUpdate];
+            var laCodeArray = new string[itemsPerUpdate];
+
+            int index = 0;
+
+            using (StreamReader reader = new StreamReader(pathToFile)) 
+            using (CsvReader csvReader = new CsvReader(reader)) {
+                csvReader.Read();
+                csvReader.ReadHeader();
+                while (csvReader.Read())
+                {
+                    var laCode = csvReader.GetField<string>("LA_Code");
+                    if (!string.IsNullOrEmpty(laCode)) 
+                    {
+                        continue;
+                    }
+                    laCode = null;
+
+                    postcodeArray[index] = csvReader.GetField("Pcode");
+                    laCodeArray[index] = laCode;
+                    index++;
+
+                    if (index == itemsPerUpdate) 
+                    {
+                        migrationBuilder.UpdateData(
+                            table: "PostcodeLookup",
+                            keyColumn: "Postcode",
+                            keyValues: postcodeArray,
+                            column: "LocalAuthorityCode",
+                            values: laCodeArray
+                        );

```

Comment:
As this is (potentially) acting on data already present, and I assume wanting to override all data previously.
This will match postcode to postcode then update all the other fields.

Unless we want any non-matches (are there any non-matches?), and for the sake of performance we might want to  explicitly truncate the table as the start of this migration? 

E,g, `migrationBuilder.Sql("TRUNCATE TABLE PostcodeLookup")`

Reply:
Though might be worthwhile checking if any remnants of reducePostcode exist before doing anything - as a fresh db should run the updated initial seed, and then not need to update anything.

Reply:
The check-if-needed does seem worthwhile!

As for truncating  - I don't think that's correct, given we're only updating the postcodes where `LA` is `null`. @ffarmanov  - should we make that a bit more explicit (a comment perhaps), it feels like the `continue` logic doesn't quite explain itself at the top it seems

Reply:
Update works fine with data which is not present, unless I want to Insert missing data